services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/SwapEndpointSpec.groovy

```diff
@@ -0,0 +1,887 @@
+
+                getConflictingFlow(flow1SwitchPair, flow1, "source", changePropertyValue(
+                        changePropertyValue(flow1.source, "portNumber", flow2.destination.portNumber), "vlanId", null)),
+                getConflictingFlow(flow1SwitchPair, flow1, "destination", changePropertyValue(
+                        changePropertyValue(flow1.destination, "portNumber", flow2.source.portNumber), "vlanId", null)),
+
+                getConflictingFlow(flow1SwitchPair, flow1, "source",
+                        changePropertyValue(flow1.source, "portNumber", flow2.destination.portNumber)),
+                getConflictingFlow(flow1SwitchPair, flow1, "destination",
+                        changePropertyValue(flow1.destination, "portNumber", flow2.source.portNumber))
+        ]
+        [flow1Src, flow1Dst, flow2Src, flow2Dst] << [
+                [changePropertyValue(changePropertyValue(flow1.source, "portNumber",
+                        flow2.destination.portNumber), "vlanId", flow2.destination.vlanId), flow1.destination,
+                 flow2.source, changePropertyValue(changePropertyValue(flow2.destination, "portNumber",
+                        flow1.source.portNumber), "vlanId", flow1.source.vlanId)],
+                [flow1.source, changePropertyValue(changePropertyValue(flow1.destination, "portNumber",
+                        flow2.source.portNumber), "vlanId", flow2.source.vlanId),
+                 changePropertyValue(changePropertyValue(flow2.source, "portNumber",
+                         flow1.destination.portNumber), "vlanId", flow1.destination.vlanId), flow2.destination],
+
+                [changePropertyValue(flow1.source, "vlanId", flow2.destination.vlanId), flow1.destination,
+                 flow2.source, changePropertyValue(flow2.destination, "vlanId", flow1.source.vlanId)],
+                [flow1.source, changePropertyValue(flow1.destination, "vlanId", flow2.source.vlanId),
+                 changePropertyValue(flow2.source, "vlanId", flow1.destination.vlanId), flow2.destination],
+
+                [changePropertyValue(flow1.source, "portNumber", flow2.destination.portNumber), flow1.destination,
+                 flow2.source, changePropertyValue(flow2.destination, "portNumber", flow1.source.portNumber)],
+                [flow1.source, changePropertyValue(flow1.destination, "portNumber", flow2.source.portNumber),
+                 changePropertyValue(flow2.source, "portNumber", flow1.destination.portNumber), flow2.destination],
+
+                [changePropertyValue(flow1.source, "portNumber", flow2.destination.portNumber), flow1.destination,
+                 flow2.source, changePropertyValue(flow2.destination, "portNumber", flow1.source.portNumber)],
+                [flow1.source, changePropertyValue(flow1.destination, "portNumber", flow2.source.portNumber),
+                 changePropertyValue(flow2.source, "portNumber", flow1.destination.portNumber), flow2.destination]
+        ]
```

Comment:
The above is quite difficult to digest, not sure whether you saved that much space compared to 'data maps' approach.